docs/tutorial/host_configs.rst

```diff
@@ -0,0 +1,239 @@
+.. # Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+.. # other BLT Project Developers. See the top-level LICENSE file for details
+.. # 
+.. # SPDX-License-Identifier: (BSD-3-Clause)
+
+.. _HostConfigs:
+
+Host-configs
+============
+
+To capture (and revision control) build options, third party library paths, etc.,
+we recommend using CMake's initial-cache file mechanism. This feature allows you
+to pass a file to CMake that provides variables to bootstrap the configuration
+process. 
+
+You can pass initial-cache files to cmake via the ``-C`` command line option.
+
+.. code-block:: bash
+
+    cmake -C config_file.cmake
+
+
+We call these initial-cache files ``host-config`` files since we typically create
+a file for each platform or for specific hosts, if necessary. 
+
+These files use standard CMake commands. CMake ``set()`` commands need to specify
+``CACHE`` as follows:
+
+.. code-block:: cmake
+
+    set(CMAKE_VARIABLE_NAME {VALUE} CACHE PATH "")
+
+Here is a snippet from a host-config file that specifies compiler details for
+using gcc 4.9.3 on LLNL's Pascal cluster. 
```

Comment:
Minor: The fact that this is `gcc@4.9.3` is not apparent from the included code since it just uses `/usr/tce/bin/gcc`.
Should we change it as follows:
```suggestion
using a custom gcc (`gcc@4.9.3`) on the LLNL Pascal cluster. 
```